From 55e417f02ad6ed68c49cc2cb8cd92f82d7d639fc Mon Sep 17 00:00:00 2001
From: Valentine Barshak <vbarshak@dev.rtsoft.ru>
Date: Fri, 25 Jan 2013 21:19:55 +0400
Subject: [PATCH 028/127] arm: tegra: ktt30: Enable 3G minicard power

This sets GPIO L5 (VI_D7) high to switch USB lane from rear IO to
3G connector and enables 3G minicard power, using GPIO D6 (VI_VSYNC).

Signed-off-by: Valentine Barshak <vbarshak@dev.rtsoft.ru>
---
 arch/arm/mach-tegra/board-ktt30-pinmux.c |    2 +-
 arch/arm/mach-tegra/board-ktt30-power.c  |    8 ++++++++
 2 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-tegra/board-ktt30-pinmux.c b/arch/arm/mach-tegra/board-ktt30-pinmux.c
index 3ab76b1..aca0141 100644
--- a/arch/arm/mach-tegra/board-ktt30-pinmux.c
+++ b/arch/arm/mach-tegra/board-ktt30-pinmux.c
@@ -237,7 +237,7 @@ static __initdata struct tegra_pingroup_config ktt30_pinmux_common[] = {
 	/* ETH_EN (Intel 82574L chip) */
 	DEFAULT_PINMUX(VI_D4,           RSVD1,           PULL_UP,   NORMAL,     OUTPUT),
 	DEFAULT_PINMUX(VI_D5,           RSVD1,           NORMAL,    NORMAL,     INPUT),
-	DEFAULT_PINMUX(VI_D7,           RSVD1,           NORMAL,    NORMAL,     OUTPUT),
+	DEFAULT_PINMUX(VI_D7,           RSVD1,           PULL_UP,   NORMAL,     OUTPUT),
 	DEFAULT_PINMUX(VI_D10,          RSVD1,           NORMAL,    NORMAL,     INPUT),
 
 	DEFAULT_PINMUX(UART3_TXD,       UARTC,           NORMAL,    NORMAL,     OUTPUT),
diff --git a/arch/arm/mach-tegra/board-ktt30-power.c b/arch/arm/mach-tegra/board-ktt30-power.c
index b534de7..997a0f1 100644
--- a/arch/arm/mach-tegra/board-ktt30-power.c
+++ b/arch/arm/mach-tegra/board-ktt30-power.c
@@ -424,6 +424,12 @@ static struct regulator_consumer_supply fixed_reg_en_vdd_bl_supply[] = {
 	REGULATOR_SUPPLY("vdd_backlight1", NULL),
 };
 
+/* EN_3V3_MODEM from AP GPIO VI_VSYNCH D06*/
+static struct regulator_consumer_supply fixed_reg_en_3v3_modem_supply[] = {
+	REGULATOR_SUPPLY("vdd_3v3_mini_card", NULL),
+	REGULATOR_SUPPLY("vdd_mini_card", NULL),
+};
+
 /* EN_VDD_PNL1 from AP GPIO VI_D6 L04*/
 #ifndef CONFIG_TEGRA_KTT30_DSI
 static struct regulator_consumer_supply fixed_reg_en_vdd_pnl1_supply[] = {
@@ -525,6 +531,7 @@ static struct regulator_consumer_supply fixed_reg_en_vddio_vid_oc_supply[] = {
 FIXED_REG(1, en_5v0,		en_5v0,		NULL,				0,      0,      TPS6591X_GPIO_2,	true,	0, 5000);
 FIXED_REG(2, en_ddr,		en_ddr,		NULL,				1,      0,      TPS6591X_GPIO_7,	true,	1, 1500);
 FIXED_REG(3, en_3v3_sys,	en_3v3_sys,	NULL,				0,      0,      TPS6591X_GPIO_6,	true,	1, 3300);
+FIXED_REG(5, en_3v3_modem,	en_3v3_modem,	NULL,				1,      0,      TEGRA_GPIO_PD6,		true,	1, 3300);
 FIXED_REG(9, en_3v3_fuse,	en_3v3_fuse,	FIXED_SUPPLY(en_3v3_sys),	0,      0,      TEGRA_GPIO_PL6,		true,	0, 3300);
 FIXED_REG(10, en_3v3_emmc,	en_3v3_emmc,	FIXED_SUPPLY(en_3v3_sys),	1,      0,      TEGRA_GPIO_PD1,		true,	1, 3300);
 FIXED_REG(11, en_vdd_sdmmc1,	en_vdd_sdmmc1,	FIXED_SUPPLY(en_3v3_sys),	0,      0,      TEGRA_GPIO_PD7,		true,	1, 3300);
@@ -554,6 +561,7 @@ FIXED_REG_OD(17, en_vddio_vid_oc,	en_vddio_vid_oc,	FIXED_SUPPLY(dis_5v_switch),
 	ADD_FIXED_REG(en_5v0),			\
 	ADD_FIXED_REG(en_ddr),			\
 	ADD_FIXED_REG(en_3v3_sys),		\
+	ADD_FIXED_REG(en_3v3_modem),		\
 	ADD_FIXED_REG(en_3v3_fuse),		\
 	ADD_FIXED_REG(en_3v3_emmc),		\
 	ADD_FIXED_REG(en_vdd_sdmmc1),		\
-- 
1.7.4.4

